felix/bpf-gpl/tc.c

```diff
@@ -239,6 +239,19 @@ static CALI_BPF_INLINE void calico_tc_process_ct_lookup(struct cali_tc_ctx *ctx)
 {
 	CALI_DEBUG("conntrack entry flags 0x%x\n", ctx->state->ct_result.flags);
 
+	/* We are forwarding a packet if it has a seen mark (that is another
+	 * program has seen it already) and is either not routed through the
+	 * bpfnat iface (which may be true for host traffic) or has the specific
+	 * reasons set.
+	 */
+	bool forwarding = CALI_F_EGRESS &&
```

Comment:
Just wondering why not defining this when it's used. Does the packet mark changes from here to the place it's used?

Reply:
my reasoning was that we do it before we start mangling the marks ourselves, so the sooner the better, but yeah sounds like a lame reason :sweat_smile: :grimacing: